### ISSUE TITLE
Severe chunk loading issues in some cases due to the wrong vertex format being used for terrain

### ISSUE THREAD

**Issue by notahumanthatisimportant** [IrisShaders/Iris#1330]
### What happened?

When changing servers on hypixel with Complementary Shaders the block textures do not load

### Screenshots

![2022-03-19_19 38 19](https://user-images.githubusercontent.com/77587691/159146679-3aaf0bad-bee7-4c16-8458-a0a125eabb0c.png)
![2022-03-19_19 38 26](https://user-images.githubusercontent.com/77587691/159146683-5c04536b-3ed5-4929-b759-fed1384935d5.png)
![2022-03-19_19 38 36](https://user-images.githubusercontent.com/77587691/159146685-2605d980-c37c-44ee-9d1f-51be9b75184b.png)

### Minecraft Version

1.18.2

### Iris Version

1.2.2-build.32

### Sodium Version

0.4.1+build.15

### Operating System

Windows 11

### What is your GPU?

NVIDIA GeForce GTX 1650 Ti

### Java Version

Java 17

**Comment by ForagedCake** [comment]
I found a temporary solution to this, you can press "F3 and T" to reload the shaderpack every time you join a lobby

**Comment by IMS212** [comment]
Just pressing K twice is enough.

**Comment by IMS212** [comment]
we’re not sure of the cause though

**Comment by Type-32** [comment]
I'm having the same problem but instead of some textures loading all of the textures just simply doesn't load. And the world also has some weird black and rgb square things.
The game log jumps out something like this: "null:1:1: error: Unknown preprocessor directive"

**Comment by IMS212** [comment]
This isn’t a texture issue, the vertex positions themselves are being corrupted for an unknown reason.

**Comment by Justsnoopy30** [comment]
Nor is it hypixel specific, I've reproduced this on a different bungeecoord server too.

**Comment by DonFlymoor** [comment]
Thanks for the shortcut, reloading shader on joining a new server could be a temporary solution, togglable in the settings

**Comment by KerbCraft** [comment]
if in the Sodium options you go to "Advanced" and change Async to Swap it seems to fix it but it makes it more laggy

**Comment by maximumpower55** [comment]
> if in the Sodium options you go to "Advanced" and change Async to Swap it seems to fix it but it makes it more laggy

You shouldn't relate on that since swap is not going to be in the next sodium version

**Comment by KerbCraft** [comment]
ik just for now it works tho

**Comment by maximumpower55** [comment]
Reloading the shader is a lot better than that fix

**Comment by mio9** [comment]
Same issue happening to me as well, although I only reload all chunks with F3+A , and it fixed the issue fast everytime

**Comment by coderbot16** [comment]
This seems to still be an issue, but just way less common / consistent. More investigation is needed.

**Comment by coderbot16** [comment]
Definitely still present in the latest version of the 1.18.2 branch, reproduced in singleplayer. It can happen when loading a world or changing settings.

![image](https://user-images.githubusercontent.com/22420959/166117421-da44d1eb-8f28-4472-94a2-5e3cd21dcef9.png)

**Comment by coderbot16** [comment]
This should finally be fully fixed now. Even after the previous fix it was still possible to reproduce this by triggering a shader reload while in a world at the right time, but now that code path has been fixed up & I've audited that there shouldn't be any other code paths to trigger this behavior. Nevertheless, the vertex format code has been modified so that even if for some reason the problematic code that triggered this previously were still active, it would not break like this.
